rir/src/compiler/pir/instruction.h

```diff
@@ -1828,7 +1827,7 @@ class FLI(RecordDeoptReason, 1, Effects::Any()) {
   public:
     DeoptReason reason;
     RecordDeoptReason(const DeoptReason& r, Value* value)
-        : FixedLenInstruction(PirType::voyd(), {{PirType::any()}}, {{value}}),
+        : FixedLenInstruction(PirType::voyd(), {{value->type}}, {{value}}),
```

Comment:
I like that change, if it works, as it should prevent us from wrongly replacing values with something of a different type..